src/common/bugsplat-api-client.ts

```diff
@@ -1,71 +1,64 @@
 import fetchPonyfill from 'fetch-ponyfill';
 import FormData from 'form-data';
+import { ApiClient, Environment } from '.';
+import { BugSplatResponse } from './api-client';
 
-export class BugSplatApiClient { 
-    private _cookie = '';
-    private _xsrfToken = '';
-    private _fetch = fetchPonyfill().fetch;
+export class BugSplatApiClient implements ApiClient {
     private _createFormData = () => new FormData();
-    
+    private _fetch = fetchPonyfill().fetch;
+    private _headers = {};
+
     constructor(
-        private _email: string,
-        private _password: string,
         private _host: string = 'https://app.bugsplat.com',
+        private _environment: Environment = Environment.Node
     ) { }
 
     createFormData(): FormData {
         return this._createFormData();
     }
 
-    async fetch(route: string, init: RequestInit = {}): Promise<Response> {
-        if (!this._cookie || !this._xsrfToken) {
-            await this.login(this._email, this._password);
-        }
-
+    async fetch(route: string, init: RequestInit = {}): Promise<BugSplatResponse> {
         if (!init.headers) {
             init.headers = {};
         }
 
-        init.headers['cookie'] = this._cookie;
+        if (this._environment === Environment.Browser) {
+            init.credentials = 'include';
+        }
 
-        if (init.method?.toUpperCase() !== 'GET') {
```

Comment:
This added seemingly unnecessary complexity. If we find out this is necessary we can always re-add it.